gengokit/gengokit_test.go

```diff
@@ -0,0 +1,432 @@
+	const def = `
+		syntax = "proto3";
+
+		// General package
+		package general;
+
+		import "google/api/annotations.proto";
+
+		// RequestMessage is so foo
+		message RequestMessage {
+		  string input = 1;
+		}
+
+		// ResponseMessage is so bar
+		message ResponseMessage {
+		  string output = 1;
+		}
+
+		// ProtoService is a service
+		service ProtoService {
+		  // ProtoMethod is simple. Like a gopher.
+		  rpc ProtoMethod (RequestMessage) returns (ResponseMessage) {
+			// No {} in path and no body, everything is in the query 
+			option (google.api.http) = { 
+				get: "/route"
+			};
+		  }
+		}
+	`
+
+	const goImportPath = "github.com/TuneLab/go-truss/gengokit"
+
+	dt, err := deftree.NewFromString(def)
+	if err != nil {
+		t.Error(err)
+	}
+
+	te, err := newTemplateExecutor(dt, goImportPath)
+	if err != nil {
+		t.Error(err)
+	}
+
+	end, err := applyTemplateFromPath("NAME-service/generated/endpoints.gotemplate", te)
+	if err != nil {
+		t.Error(err)
+	}
+
+	endCode, err := ioutil.ReadAll(end)
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = formatCode(endCode)
+	if err != nil {
+		t.Error(err)
+	}
+
+}
+
+func TestUpdateServerMethods(t *testing.T) {
+	const def = `
+		syntax = "proto3";
+
+		// General package
+		package general;
+
+		import "google/api/annotations.proto";
+
+		// RequestMessage is so foo
+		message RequestMessage {
+		  string input = 1;
+		}
+
+		// ResponseMessage is so bar
+		message ResponseMessage {
+		  string output = 1;
+		}
+
+		// ProtoService is a service
+		service ProtoService {
+		  // ProtoMethod is simple. Like a gopher.
+		  rpc ProtoMethod (RequestMessage) returns (ResponseMessage) {
+			// No {} in path and no body, everything is in the query 
+			option (google.api.http) = { 
+				get: "/route"
+			};
+		  }
+		}
+	`
+
+	const goImportPath = "github.com/TuneLab/go-truss/gengokit"
+
+	dt, err := deftree.NewFromString(def)
+	if err != nil {
+		t.Error(err)
+	}
+
+	te, err := newTemplateExecutor(dt, goImportPath)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// apply server_handler.go template
+	sh, err := applyTemplateFromPath("NAME-service/handlers/server/server_handler.gotemplate", te)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// read the code off the io.Reader
+	shCode, err := ioutil.ReadAll(sh)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// format the code
+	shCode, err = formatCode(shCode)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// updateServerMethods with the same templateExecutor
+	same, err := updateServerMethods(bytes.NewReader(shCode), te)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// read the new code off the io.Reader
+	sameCode, err := ioutil.ReadAll(same)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// format that new code
+	sameCode, err = formatCode(sameCode)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// make sure the code is the same
+	if bytes.Compare(shCode, sameCode) != 0 {
+		t.Fatalf("\n__BEFORE__\n\n%v\n\n__AFTER\n\n%v\n\nCode before and after updating differs",
```

Comment:
I've noticed there are a lot of `%v`s in the Truss code where there would more appropriately be `%s`s. It's not a problem in terms of execution, but it's stylistically odd.